multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -165,842 +158,159 @@ fn payments_were_adjusted_due_to_insufficient_balances() {
-        gwei_to_wei::<u128, _>(payment_thresholds.debt_threshold_gwei + 60_000_000);
     // Assuming all Nodes rely on the same set of payment thresholds
-    let cons_node_initial_service_fee_balance_minor = (owed_to_serv_node_1_minor
-        + owed_to_serv_node_2_minor)
-        - gwei_to_wei::<u128, u64>(2_345_678);
-    let agreed_transaction_fee_unit_price_major = 60;
-    let transaction_fee_needed_to_pay_for_one_payment_major = {
-        // We will need less but this should be accurate enough
-        let txn_data_with_maximized_cost = [0xff; 68];
+    let owed_to_serv_node_1_minor = to_wei(payment_thresholds.debt_threshold_gwei + 5_000_000);
+    let owed_to_serv_node_2_minor = to_wei(payment_thresholds.debt_threshold_gwei + 20_000_000);
+    // Account of Node 3 will be a victim of tx fee insufficiency and will fall away, as its debt
+    // is the heaviest, implying the smallest weight evaluated and the last priority compared to
+    // those two others.
+    let owed_to_serv_node_3_minor = to_wei(payment_thresholds.debt_threshold_gwei + 60_000_000);
+    let enough_balance_for_serving_node_1_and_2 =
+        owed_to_serv_node_1_minor + owed_to_serv_node_2_minor;
+    let consuming_node_initial_service_fee_balance_minor =
+        enough_balance_for_serving_node_1_and_2 - to_wei(2_345_678);
+    let gas_price_major = 60;
+    let tx_fee_needed_to_pay_for_one_payment_major = {
+        // We'll need littler funds, but we can stand mild inaccuracy from assuming the use of
+        // all nonzero bytes in the data in both txs, which represents maximized costs
+        let txn_data_with_maximized_costs = [0xff; 68];
         let gas_limit_dev_chain = {
             let const_part = web3_gas_limit_const_part(Chain::Dev);
             u64::try_from(compute_gas_limit(
                 const_part,
-                txn_data_with_maximized_cost.as_slice(),
+                txn_data_with_maximized_costs.as_slice(),
             ))
             .unwrap()
         };
-        let transaction_fee_margin = Percentage::new(15);
-        transaction_fee_margin
-            .add_percent_to(gas_limit_dev_chain * agreed_transaction_fee_unit_price_major)
+        let transaction_fee_margin = PurePercentage::try_from(15).unwrap();
+        transaction_fee_margin.add_percent_to(gas_limit_dev_chain * gas_price_major)
     };
-    eprintln!(
-        "Computed transaction fee: {}",
-        transaction_fee_needed_to_pay_for_one_payment_major
-    );
-    let cons_node_transaction_fee_balance_minor =
-        2 * gwei_to_wei::<u128, u64>(transaction_fee_needed_to_pay_for_one_payment_major);
-    let test_global_config = TestInputsOutputsConfig {
-        ui_ports_opt: Some(Ports {
-            consuming_node: consuming_node_ui_port,
-            serving_node_1: serving_node_1_ui_port,
-            serving_node_2: serving_node_2_ui_port,
-            serving_node_3: serving_node_3_ui_port,
-        }),
+    const AFFORDABLE_PAYMENTS_COUNT: u128 = 2;
```

Comment:
You may migrate this constant to the top, maybe along with a new constant for `GAS_PRICE_MARGIN` (only in this test). I think once you do that you might not need to introduce the comment for affordable payments.